### PR TITLE
Added ShizaProject semi-private tracker

### PR DESCRIPTION
**Pull request by dm3ch**

### CODE REVIEW

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+            try
+            {
+                var parser = new HtmlParser();
+                var document = await parser.ParseDocumentAsync(result.ContentString);
+
+                foreach (var linkNode in document.QuerySelectorAll(ReleaseLinksSelector))
+                {
+                    var url = linkNode.GetAttribute("href");
+                    releases.AddRange(await FetchShowReleases(url));
```
Comment (dm3ch) [inline src/Jackett.Common/Indexers/ShizaProject.cs:117]: The only problem I have faced for now, is that when we need to process multiple releases it took more time than it's comfortable.
You can check that if you search empty string.
The problem itself that release page request took 1-2 seconds and they are sequential.

I wonder if someone could consult me how to make pages requests parallel.

**Comment by garfield69** [comment]
@ngosang  is our local C# dev and will do a code review for code compliance and can answer your C# questions.
He has a lot on his plate ATM so be patient.
While I'm not a C# dev, I can make this observation: I don't think we want to do parallel queries, we need to avoid appearing like a DDoS attack, which  most web sites are quite wary off these days.

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                   name: "ShizaProject",
+                   description: "ShizaProject Tracker is a semi-private russian tracker and release group for anime",
+                   link: "http://shiza-project.com/",
+                   caps: new TorznabCapabilities(),
+                   configService: configService,
+                   client: wc,
+                   logger: l,
+                   p: ps,
+                   configData: new ConfigurationDataBasicLogin())
+        {
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/ShizaProject.cs:33]: given you have to login with email and password you should use ConfigurationDataBasicLoginWithEmail

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            Encoding = Encoding.UTF8;
+            Language = "ru-ru";
+            Type = "semi-private";
+        }
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/ShizaProject.cs:37]: you need to define a category
```suggestion
        }            AddCategoryMapping(1, TorznabCatType.TVAnime, "Anime");

```

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            set => configData = value;
+        }
+
+        /// <summary>
+        /// http://shiza-project.com/accounts/login
+        /// </summary>
+        private string LoginUrl => SiteLink + "accounts/login";
+
+        /// <summary>
+        /// https://tr.ShizaProject.com/index.php?do=search
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/ShizaProject.cs:51]: that comment looks dodgy

**Review by garfield69 — CHANGES_REQUESTED** [review]

File: src/Jackett.Common/Indexers/ShizaProject.cs (hunk 4 above)
Comment (dm3ch) [reply src/Jackett.Common/Indexers/ShizaProject.cs]: TY, missed that

File: src/Jackett.Common/Indexers/ShizaProject.cs (hunk 2 above)
Comment (dm3ch) [reply src/Jackett.Common/Indexers/ShizaProject.cs:33]: Migrated

**Comment by dm3ch** [comment]
@ngosang Fixed code issues you found during review, TY

**Review by garfield69 — APPROVED** [review]
tested. LGTM

**Review by ngosang — APPROVED** [review]
> While I'm not a C# dev, I can make this observation: I don't think we want to do parallel queries, we need to avoid appearing like a DDoS attack, which most web sites are quite wary off these days.

The code LGTM but I'm concerning too about the number of requests we are making. If this is the only way it's ok, but the the site admin complains we should remove this indexer.

**Review by ngosang — CHANGES_REQUESTED** [review]
Apply changes from this PR #9885 into this. The GUID is mandatory and must be unique. It's used for RSS. In most cases the link or magnet can be used.

**Comment by dm3ch** [comment]
> > While I'm not a C# dev, I can make this observation: I don't think we want to do parallel queries, we need to avoid appearing like a DDoS attack, which most web sites are quite wary off these days.
> 
> The code LGTM but I'm concerning too about the number of requests we are making. If this is the only way it's ok, but the the site admin complains we should remove this indexer.

So for now we are 

> Apply changes from this PR #9885 into this. The GUID is mandatory and must be unique. It's used for RSS. In most cases the link or magnet can be used.

Added GUID.

**Review by ngosang — APPROVED** [review]

**Comment by garfield69** [comment]
Jackett 0.16.1819

**Pull request merged**